src/neo4j/mod.rs

```diff
@@ -0,0 +1,702 @@
+/// // connect to Neo4j with the uri, user and pass
+/// ```
+///
+/// # Neo4j Version
+///
+/// The version of the image can be set with the `NEO4J_VERSION_TAG` environment variable.
+/// The default version is `5`.
+/// The available versions can be found on [Docker Hub](https://hub.docker.com/_/neo4j/tags).
+///
+/// The used version can be retrieved with the `version` method.
+///
+/// # Auth
+///
+/// The default user is `neo4j` and the default password is `neo`.
+///
+/// The used user can be retrieved with the `user` method.
+/// The used password can be retrieved with the `pass` method.
+///
+/// # Environment variables
+///
+/// The following environment variables are supported:
+///   * `NEO4J_VERSION_TAG`: The default version of the image to use.
+///   * `NEO4J_TEST_USER`: The default user to use for authentication.
+///   * `NEO4J_TEST_PASS`: The default password to use for authentication.
+///
+/// # Neo4j Labs Plugins
+///
+/// Neo4j offers built-in support for Neo4j Labs plugins.
+/// The method `with_neo4j_labs_plugin` can be used to define them.
+///
+/// Supported plugins are APOC, APOC Core, Bloom, Streams, Graph Data Science, and Neo Semantics.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Neo4j {
+    version: Value,
+    user: Value,
+    pass: Value,
+    enterprise: bool,
+    plugins: Vec<Neo4jLabsPlugin>,
+}
+
+impl Neo4j {
+    const DEFAULT_USER: &'static str = "neo4j";
+    const DEFAULT_PASS: &'static str = "neo";
+    const DEFAULT_VERSION_TAG: &'static str = "5";
+
+    /// Create a new instance of a Neo4j image.
+    #[must_use]
+    pub const fn new() -> Self {
+        Self {
+            version: Value::Default(Self::DEFAULT_VERSION_TAG),
+            user: Value::Default(Self::DEFAULT_USER),
+            pass: Value::Default(Self::DEFAULT_PASS),
+            enterprise: false,
+            plugins: Vec::new(),
+        }
+    }
+
+    /// Create a new instance of a Neo4j 5 image with the default user and password.
+    #[must_use]
+    pub const fn from_env() -> Self {
+        Self {
+            version: Value::Env {
+                var: "NEO4J_VERSION_TAG",
+                fallback: Self::DEFAULT_VERSION_TAG,
+            },
+            user: Value::Env {
+                var: "NEO4J_TEST_USER",
+                fallback: Self::DEFAULT_USER,
+            },
+            pass: Value::Env {
+                var: "NEO4J_TEST_PASS",
+                fallback: Self::DEFAULT_PASS,
+            },
+            enterprise: false,
+            plugins: Vec::new(),
+        }
+    }
+
+    /// Set the Neo4j version to use.
+    /// The value must be an existing Neo4j version tag.
+    ///
+    /// Only a subset of Semantic Versions are supported.
+    /// The version must be of the format
+    ///
+    ///    MAJOR[.MINOR[.PATCH]]
+    ///
+    ///
+    /// # Errors
+    ///
+    /// If the version is not valid according to the format.
+    pub fn with_version(
```

Comment:
I'm asking myself whether this check is really worth the additional dependency. If you provide a non existing tag the container startup will fail anyways.

Another question I ask myself is if we really want to force the user to use the official image :thinking:

@DDtKey WDYT?

Reply:
Agree, I think we should avoid version check. 
Users may want to use their own registries with custom tagging (not such a rare case)

Reply:
Ok, good point. I'll remove the validation and the extra dependency.